18F/dol-whd-14c
# Directive for anchor links

#### What's this PR do?
- Implements AngularJS header component over Angular(4) component
- Fixes "Skip to Main Content" anchor link with new anchor-link directive
- adds keypress functionality to anchor-link

#### How should this be manually tested?
- Visit site, click address bar, hit "TAB" on your keyboard, and press "Skip to Main Content" link that appears. Verify that the page scrolls AND focuses on mainContent div

@ltsheu can you verify I did the css correctly for this? Let me know if you need help viewing this before it is merged.

#### Any background context you want to provide?
TLDR: I was running into issues with karma when trying to use an Angular4 header component so I used an AngularJS header component.

Angular 4 doesn’t support anchor links all that well out of the box -- at least not intuitively (see https://github.com/angular/angular/issues/6595).  I have implemented a fix in a separate branch that uses an Angular(4) header component, but only at the expense of running the AngularJS and Angular(4) routers side by side as well as messing around with some of the webpack settings. However, because of this and some other modifications to get the app working properly as a hybrid app, the karma tests are not able to run properly on the AngularJS components. I need to do some more investigation in how to test both Angular and AngularJS components -- I suggest we assign another user story for this and assess its priority separately.

Since I didn't want to block the user story pertaining to this issue, I went ahead and used an AngularJS header component instead. I spoke to @mmurthydol about this and we agreed upon this path forward. In case you are interested, the branch that uses the Angular4 header component can be found here: 
https://github.com/18F/dol-whd-14c/tree/fix/ngupgrade.

#### What are the relevant tickets?
#307, #336 

#### Screenshots (if appropriate)
#### Questions:
